sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -223,6 +223,8 @@ class Analyzer(
           case other => Alias(other, other.toString)()
         }
 
+        val attributeMap = groupByAliases.map(a => (a -> a.toAttribute.withNullability(true))).toMap
```

Comment:
This can be merged first. I will add a following one for the issue later. If it is ok for you.